BUG: test_rint_big_int fails on Core 2 Duo CPU due to precision

### Describe the issue:

Building Numpy 1.23.1 with GNU Guix on an old Core 2 Duo (Intel Q6700) CPU, I stumbled on the `test_rint_big_int` failure (see below for the error message).

The error seems to be related to precision, perhaps specific to this x86_64 architecture 

### Reproduce the code example:

```python
It seems to be reproducible every time I build on this Core 2 Duo machine.
```
Reproducible by disabling CPU feature `SSE41` on runtime via:

```Bash
export NPY_DISABLE_CPU_FEATURES="SSE41"
```


### Error message:

```shell
=================================== FAILURES ===================================
______________________________ test_rint_big_int _______________________________
[gw0] linux -- Python 3.9.13 /gnu/store/ir68khi30jzdlcn9sjw0anlgsgsjfjfi-python-3.9.13/bin/python3

    def test_rint_big_int():
        # np.rint bug for large integer values on Windows 32-bit and MKL
        # https://github.com/numpy/numpy/issues/6685
        val = 4607998452777363968
        # This is exactly representable in floating point
        assert_equal(val, int(float(val)))
        # Rint should not change the value
>       assert_equal(val, np.rint(val))
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: 4607998452777363968
E        DESIRED: 4.6079984527773635e+18

val        = 4607998452777363968

/gnu/store/bg6dygk3x2dpc15s40k61hy37nzhaiw2-python-numpy-1.23.1/lib/python3.9/site-packages/numpy/core/tests/test_umath.py:3875: AssertionError
=========================== short test summary info ============================
FAILED ../../core/tests/test_umath.py::test_rint_big_int - AssertionError: 
= 1 failed, 17805 passed, 1370 skipped, 35 xfailed, 7 xpassed in 474.41s (0:07:54) =
error: in phase 'check': uncaught exception:
%exception #<&invoke-error program: "./runtests.py" arguments: ("-vv" "--no-build" "--mode=fast" "-j" "4" "--" "-n" "4" "-k" "not test_float_remainder_overflow and not test_pareto") exit-status: 1 term-signal: #f stop-signal: #f> 
phase `check' failed after 476.8 seconds
```

### NumPy/Python version information:

1.23.1

### Context for the issue:

Occurs while running the test suite during packaging.

EDIT(Sayed): Reproduce

--- a/numpy/core/src/common/simd/sse/math.h
+++ b/numpy/core/src/common/simd/sse/math.h
@@ -287,14 +287,16 @@ NPY_FINLINE npyv_f64 npyv_rint_f64(npyv_f64 a)
 #else
     const npyv_f64 szero = _mm_set1_pd(-0.0);
     const npyv_f64 two_power_52 = _mm_set1_pd(0x10000000000000);
-    npyv_f64 sign_two52 = _mm_or_pd(two_power_52, _mm_and_pd(a, szero));
+    npyv_f64 abs_a = npyv_abs_f64(a);
     // round by add magic number 2^52
-    npyv_f64 round = _mm_sub_pd(_mm_add_pd(a, sign_two52), sign_two52);
-    // respect signed zero, e.g. -0.5 -> -0.0
-    return _mm_or_pd(round, _mm_and_pd(a, szero));
+    // assuming that MXCSR register is set to rounding
+    npyv_f64 abs_round = _mm_sub_pd(_mm_add_pd(two_power_52, abs_a), two_power_52);
+    // copysign
+    npyv_f64 round = _mm_or_pd(abs_round, _mm_and_pd(a, szero));
+    // a if a >= 2^52
+    return npyv_select_f64(npyv_cmpge_f64(abs_a, two_power_52), a, round);
 #endif
 }
-
 // ceil
 #ifdef NPY_HAVE_SSE41
     #define npyv_ceil_f32 _mm_ceil_ps
@@ -316,10 +318,7 @@ NPY_FINLINE npyv_f64 npyv_rint_f64(npyv_f64 a)
     {
         const npyv_f64 szero = _mm_set1_pd(-0.0);
         const npyv_f64 one = _mm_set1_pd(1.0);
-        const npyv_f64 two_power_52 = _mm_set1_pd(0x10000000000000);
-        npyv_f64 sign_two52 = _mm_or_pd(two_power_52, _mm_and_pd(a, szero));
-        // round by add magic number 2^52
-        npyv_f64 round = _mm_sub_pd(_mm_add_pd(a, sign_two52), sign_two52);
+        npyv_f64 round = npyv_rint_f64(a);
         npyv_f64 ceil = _mm_add_pd(round, _mm_and_pd(_mm_cmplt_pd(round, a), one));
         // respect signed zero, e.g. -0.5 -> -0.0
         return _mm_or_pd(ceil, _mm_and_pd(a, szero));
@@ -350,7 +349,9 @@ NPY_FINLINE npyv_f64 npyv_rint_f64(npyv_f64 a)
         // round by add magic number 2^52
         npyv_f64 abs_round = _mm_sub_pd(_mm_add_pd(abs_a, two_power_52), two_power_52);
         npyv_f64 subtrahend = _mm_and_pd(_mm_cmpgt_pd(abs_round, abs_a), one);
-        return _mm_or_pd(_mm_sub_pd(abs_round, subtrahend), _mm_and_pd(a, szero));
+        npyv_f64 trunc = _mm_or_pd(_mm_sub_pd(abs_round, subtrahend), _mm_and_pd(a, szero));
+        // a if a >= 2^52
+        return npyv_select_f64(npyv_cmpge_f64(abs_a, two_power_52), a, trunc);
     }
 #endif
 

--- a/numpy/core/tests/test_simd.py
+++ b/numpy/core/tests/test_simd.py
@@ -437,6 +437,16 @@ class _SIMD_FP(_Test_Utility):
                 _round = intrin(data)
                 assert _round == data_round
 
+        # test large numbers
+        for i in (
+            1.1529215045988576e+18, 4.6116860183954304e+18,
+            5.902958103546122e+20, 2.3611832414184488e+21
+        ):
+            x = self.setall(i)
+            y = intrin(x)
+            data_round = [func(n) for n in x]
+            assert y == data_round
+
         # signed zero
         if intrin_name == "floor":
             data_szero = (-0.0,)

